### DISCUSSION TITLE
Apply condition to Video Capture or Video Writer

### DISCUSSION THREAD

**Discussion by adamhung60** [bonsai-rx/bonsai#1177] [Q&A]
I want my program to begin recording a video after e.g a Keydown node is triggered. Is this possible? The Video Writer and Video Capture Device nodes do not seem to want to cooperate with condition nodes or anything else I have tried. 

Separately, I was wondering if it would be possible to use a Fastec camera with VideoCapture Device, or what the requirements are to use the VideoCapture Device node on a certain camera, as I don't really understand how this node works.

Thanks!

**Comment by bruno-f-cruz** [comment]
Hey,
Regarding the first questions, a quick search returned some relevant posts:
#870
#868 
#1023 

Regarding the second question, `VideoCaptureDevice` uses generic DirectShow drivers. Not sure if the camera in question supports it. The best way to find out is probably to email the manufacturing company and ask.
Hope it helps!
Bruno

**Reply by adamhung60** [reply]
Thank you! Got it to work how I wanted. And I will contact the company about the camera.

Out of curiosity, is there a reason the VideoWriter can't directly take a bool or condition input? I used publish subject  
+ keydown -> select many -> subscribe subject to start recording video but I don't see why this works and the more straightforward way doesn't.

**Reply by bruno-f-cruz** [reply]
Not sure what you mean, perhaps add an image with the workflow? I am not sure what the behavior should be when you try to send a "boolean" to a videowriter? I am not sure what you are trying to record...

**Reply by adamhung60** [reply]
This is a bit different from my original question, but basically I am confused why you can't drag an arrow from the Condition into the Subscribe Subject:

<img width="264" alt="Screenshot 2022-12-09 at 2 10 20 PM" src="https://user-images.githubusercontent.com/120136367/206779320-50e081a8-2c37-46e7-9602-320b9668bb42.png">

But if you just put the Subscribe Subject -> Video writer into a Select Many, suddenly you can:

<img width="356" alt="Screenshot 2022-12-09 at 2 10 52 PM" src="https://user-images.githubusercontent.com/120136367/206779406-fd2d7668-ee10-46bb-b535-232cb9efa820.png">

**Reply by bruno-f-cruz** [reply]
ahh got it. So in your first example, the output of the condition will be the incoming boolean, which is not compatible with the videowriter.In your second example, you are using a `SelectMany`. This node does something very different when it receives an input. For the most part you can see it as starting a workflow from scratch that is encapsulated in the node. Inside, it will subscribe to the video stream and start saving it, so you don't have problems with the event type. The `SelectMany` behavior is such that, regardless of the input type, it will always start a new observable sequence.

If you want to read more about it, and get an intuition for these so-called "higher order operators" give this link a look 
https://bonsai-rx.org/docs/articles/higher-order.html

**Reply by adamhung60** [reply]
Ok that makes sense, thank you!
